scrapy/shell.py

```diff
@@ -102,25 +103,33 @@ def _schedule(self, request: Request, spider: Spider | None) -> defer.Deferred[A
             # set the asyncio event loop for the current thread
             event_loop_path = self.crawler.settings["ASYNCIO_EVENT_LOOP"]
             set_asyncio_event_loop(event_loop_path)
-        spider = self._open_spider(request, spider)
+
+        def crawl_request(_):
+            assert self.crawler.engine is not None
+            self.crawler.engine.crawl(request)
+
+        d2 = self._open_spider(request, spider)
+        d2.addCallback(crawl_request)
+
         d = _request_deferred(request)
         d.addCallback(lambda x: (x, spider))
-        assert self.crawler.engine
-        self.crawler.engine.crawl(request)
```

Comment:
I needed this to ensure that the spider had been opened by the time `engine.crawl()` was called.